### PR TITLE
HIP-786

### LINKED ISSUE #8535: Port new staking prop names to mod `StakingService`
### Problem

Need to port https://github.com/hashgraph/hedera-services/pull/8247 to mod code.

### Solution

Do so.

### Alternatives

_No response_

### LINKED ISSUE #8549: Implement HIP-786
### Problem

We need to set the new `NodeStakeUpdateTransactionBody` fields as prescribed by [HIP-786](https://hips.hedera.com/hip/hip-786).

### Solution

Do this.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Reverts unintentional change of `NftID#tokenID` case to `tokenId`.
 - Updates `File` constructor usages for the [new `preSystemDeleteExpirationSecond` field](https://github.com/hashgraph/hedera-protobufs/blob/revert-nft-token-id-case/services/state/file/file.proto#L62).
 - Closes #8535
 - Closes #8549

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculator.java [hunk 1]
```diff
@@ -203,12 +203,18 @@ public void updateNodes(final Instant consensusTime) {
...
                 lastInstantOfPreviousPeriodFor(consensusTime),
                 nodeStakingInfos,
                 properties,
                 totalStakedRewardStart,
-                perHbarRate);
+                perHbarRate,
+                reservedStakingRewards,
+                unreservedStakingRewardBalance,
+                dynamicProperties.stakingRewardBalanceThreshold(),
+                dynamicProperties.maxStakeRewarded());
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculator.java:217]: Will these changes be made in the modularized code too?

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/settings.gradle.kts [hunk 2]
```diff
@@ -81,8 +81,8 @@ fun include(name: String, path: String) {
 
 // The HAPI API version to use for Protobuf sources. This can be a tag or branch
 // name from the hedera-protobufs GIT repo.
-val hapiProtoVersion = "0.40.0-blocks-state-SNAPSHOT"
-val hapiProtoBranchOrTag = "add-pbj-types-for-state" // hapiProtoVersion
+val hapiProtoVersion = "0.43.0-rc-SNAPSHOT"
+val hapiProtoBranchOrTag = "revert-nft-token-id-case" // hapiProtoVersion
```
Comment (jsync-swirlds) [inline hedera-node/settings.gradle.kts:85]: This should be a tag rather than a branch, so we have more stable builds.

**Review by JeffreyDallas — DISMISSED** [review]

**Review by qnswirlds — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by kimbor — COMMENTED** [review]
It looks like this is the commit where tokenID was changed to token_id: https://github.com/hashgraph/hedera-protobufs/commit/e1b59db760c37f6b724259c356f25dcece6087d9
In this commit, you change token_id to token_ID: https://github.com/hashgraph/hedera-protobufs/pull/306/files#diff-636b14bb9e5952ee6f3253a1bba9a76bde7e8a9a27a78c0eda2d6303ebb3bde6

Note that an underscore was added. So it went tokenID -> token_id -> token_ID

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculator.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculator.java:217]: That's done [in this PR](https://github.com/hashgraph/hedera-services/blob/329e704688cc0acd6de7b4c3da68a498b826b2c4/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/EndOfStakingPeriodUpdater.java#L478). 😁

**Comment by tinker-michaelj** [comment]
> It looks like this is the commit where tokenID was changed to token_id: [hashgraph/hedera-protobufs@e1b59db](https://github.com/hashgraph/hedera-protobufs/commit/e1b59db760c37f6b724259c356f25dcece6087d9) In this commit, you change token_id to token_ID: https://github.com/hashgraph/hedera-protobufs/pull/306/files#diff-636b14bb9e5952ee6f3253a1bba9a76bde7e8a9a27a78c0eda2d6303ebb3bde6
> 
> Note that an underscore was added. So it went tokenID -> token_id -> token_ID

This should generate all the same getters/setters though I think.

**Pull request merged**